pytorch_lightning/trainer/trainer.py

```diff
@@ -522,7 +517,10 @@ def train(self):
                     return
 
                 # update LR schedulers
-                self.optimizer_connector.update_learning_rates(interval='epoch')
+                self.optimizer_connector.update_learning_rates(
+                    interval='epoch',
+                    opt_indices=[opt_idx for opt_idx, _ in self.train_loop.get_optimizers_iterable()],
+                )
```

Comment:
I don't think it's required when updating on the epoch level. Is it?

Reply:
I think some schedulers like `torch.optim.lr_scheduler.CosineAnnealingLR` require a `max_epochs` parameter.

Reply:
in that case, the interval key in lr_dict will be `step` so it will act as a step lr_scheduler. Also, frequency works on step level.

Reply:
If the optimizers are configured as 
```
return [
                {'optimizer': optimizer1, 'frequency': 6},
                {
                    'optimizer': optimizer2,
                    'frequency': 12,
                    'lr_scheduler': {
                        'scheduler': optim.lr_scheduler.CosineAnnealingLR(optimizer2, 2),
                        'interval': 'epoch',
                    },
                },
            ]
```
The LR scheduler will be `epoch` but the frequency is still `step` I believe.

Reply:
okay yeah, saw your example. never thought frequency can be used this way too.